### ISSUE TITLE
Ordering in hasManyThrough relation doesn't work

### ISSUE THREAD

**Issue by bbrody** [spotorm/spot2#292]
I have `ProjectEntity` with such relation in it 
`'users' => $mapper->hasManyThrough($entity, 'Api\V2\Entities\UserEntity', 'Api\V2\Entities\UserProjectEntity', 'user_id', 'project_id')
                ->active()->order(['displayname' => 'ASC'])`
`->order()` does not affect on the result, its sorted by id of `UserProjectEntity` (it has simple structure `id - user_id -project_id`)

**Comment by FlipEverything** [comment]
Yeah, I don't think that will work out of the box. It's possibly a bug.

**Comment by vlucas** [comment]
I am curious about this now. What table is the column `displayname` on? Is it on either of those join tables?

**Comment by bbrody** [comment]
Hi! Any updates on this issue?

**Comment by Popoviciu-Andrei** [comment]
You can fix this by editing the eagerLoadOnCollection method in vendor/vlucas/spot2/lib/Relation/HasManyThrough.php and changing this:
foreach ($collectionRelations as $relatedEntity) {
    $relatedEntityPk = $relatedEntity->$relationRelatedForeignKey;

    if ($relatedEntityPk == $throughForeignKey) {
        $entityRelations[$throughLocalKey][] = $relatedEntity;
    }
}

to this:

foreach ($collectionRelations as $idx=>$relatedEntity) {
    $relatedEntityPk = $relatedEntity->$relationRelatedForeignKey;

    if ($relatedEntityPk == $throughForeignKey) {
        $entityRelations[$throughLocalKey][$idx] = $relatedEntity;
    }
}
ksort($entityRelations[$throughLocalKey]);
